master/internal/trial.go

```diff
@@ -384,15 +379,13 @@ func (t *trial) runningReceive(ctx *actor.Context) error {
 
 func (t *trial) processScheduler(ctx *actor.Context) error {
 	switch msg := ctx.Message().(type) {
-
 	case scheduler.TaskAssigned:
 		if err := t.processAssigned(ctx, msg); err != nil {
 			return err
 		}
 
 	case scheduler.TerminateRequest:
-		ctx.Log().Info("trial runner requested to terminate")
-		t.pendingGracefulTermination = true
+		t.terminate(ctx, false)
```

Comment:
blocking: the way `t.terminate()` is  implemented right now is that it the next step and checkpoint don't complete during the timeout, we will forcibly kill the trial, which is different from the previous behavior where we always allowed the trial to complete the step and checkpoint.